code/services/QueuedJobService.php

```diff
@@ -615,14 +615,20 @@ public function runJob($jobId) {
 
 		Config::unnest();
 
-		// okay lets reset our user if we've got an original
+		// okay let's reset our user if we've got an original
 		if ($runAsUser && $originalUser) {
 			Session::clear("loggedInAs");
 			if ($originalUser) {
 				Session::set("loggedInAs", $originalUser->ID);
 			}
 		}
 
+		// let's make sure there is a cleanupJob in the queue
+		if (Config::inst()->get('CleanupJob', 'is_enabled')) {
```

Comment:
Could the job itself be responsible for re-adding itself to the queue? The initial creation can be done directly from the UI - my thinking being that if you want to _not_ have the cleanup running, you're needing to make server level configuration changes rather than just deleting the pending job from the UI.